src/main/java/org/wildfly/security/sasl/util/SecurityProviderSaslClientFactory.java

```diff
@@ -61,7 +63,12 @@ public SecurityProviderSaslClientFactory(final Supplier<Provider[]> providerSupp
      * Construct a new instance.  The currently installed system providers are used.
      */
     public SecurityProviderSaslClientFactory() {
-        this(Security::getProviders);
+        this(AccessController.doPrivileged(new PrivilegedAction<Supplier<Provider[]>>() {
```

Comment:
This doesn't actually do anything because returning the method handle to get the security providers does not require any permissions.